docs/en/sql-reference/data-types/map.md

```diff
@@ -6,101 +6,107 @@ sidebar_label: Map(K, V)
 
 # Map(K, V)
 
-`Map(K, V)` data type stores `key:value` pairs.
-The Map datatype is implemented as `Array(Tuple(key T1, value T2))`, which means that the order of keys in each map does not change, i.e., this data type maintains insertion order.
+Data type `Map(K, V)` stores key-value pairs.
 
-**Parameters**
+Unlike other databases, maps are not unique in ClickHouse, i.e. a map can contain two elements with the same key.
+(The reason for that is that maps are internally implemented as `Array(Tuple(K, V))`.)
+
+You can use use syntax `m[k]` to obtain the value for key `k` in map `m`.
+If more than one element with key `k` exists, the value for the first element is returned.
```

Comment:
Never knew it. It looks kind of crazy. How can we get all the records then... It is possible to get different result after sort?

Reply:
I agree that the implementation is unusual. Anyway, changing it for something better seems too much work ...

There is no function which returns _all_ elements for a key ... I think maps are simply not supposed to be used as a non-unique store...